config/env_sample.hjson

```diff
@@ -252,7 +253,36 @@
                 and membership.id is not null 
                 and REGEXP_EXTRACT(`group`.id, r'.*:(.*)') IN UNNEST(@course_ids)
                 '''
-        }
+        },
+        "canvas_pages": {
+            "query":
+                '''
+                SELECT 'canvas_pages' AS resource_type,
+                CONCAT(CAST(REGEXP_EXTRACT(`group`.id, r'.*:(.*)') AS INT64) - @canvas_data_id_increment, 
+        				'/pages/', 
+        				CAST(JSON_EXTRACT_SCALAR(object.extensions, '$[\'com.instructure.canvas\'][entity_id]') as INT64) - @canvas_data_id_increment) as resource_id,
```

Comment:
this is a good idea of incorporating the course id along with page id as resource id. This solves the problem not touching the view.py code.

Reply:
The change seems to work now the resource id  `course-id/pages/page-id`. I was able to test with courses that has pages with non-ASCII character as well

Reply:
Great! Thanks for checking!

Reply:
Great! I will mark this as resolved.